agent/src/agentclient.rs

```diff
@@ -0,0 +1,173 @@
+use k8s_openapi::api::core::v1::Node;
+use kube::Api;
+use snafu::{futures::TryFutureExt, ResultExt};
+use std::env;
+use tokio::time::{sleep, Duration};
+
+use crate::apiclient::{boot_update, get_os_info, list_available, prepare, update};
+use crate::error::{self, Result};
+use apiserver::api::UpdateBottlerocketNodeRequest;
+use models::constants::NAMESPACE;
+use models::node::{
+    BottlerocketNode, BottlerocketNodeSelector, BottlerocketNodeState, BottlerocketNodeStatus,
+};
+
+const AGENT_SLEEP_DURATION: Duration = Duration::from_millis(5000);
+
+#[derive(Clone)]
+pub struct BrupopAgent {
+    k8s_client: kube::client::Client,
+}
+
+impl BrupopAgent {
+    pub fn new(k8s_client: kube::client::Client) -> Self {
+        BrupopAgent { k8s_client }
+    }
+
+    pub async fn ensure_node_custom_resource_exists(&self) {
+        // Todo (tianhg)
+    }
+
+    async fn fetch_custom_resource(&self) -> Result<BottlerocketNode> {
+        // TODO Nit: This comment doesn't say much more than the line of code following it.
+        // It might be good to add more context about what the node name is, or what it's used for.
+        let associated_bottlerocketnode_name =
```

Comment:
Nit: why not `associated_bottlerocket_node_name` to be consistent with `associated_node_name` in lines below?